homeassistant/components/balboa/config_flow.py

```diff
@@ -0,0 +1,115 @@
+"""Config flow for Balboa Spa Client integration."""
+from pybalboa import BalboaSpaWifi
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import _LOGGER, CONF_SYNC_TIME, DOMAIN
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data[CONF_HOST] == data[CONF_HOST]:
+            raise AlreadyConfigured
+
+    _LOGGER.debug("Attempting to connect to %s", data[CONF_HOST])
+    spa = BalboaSpaWifi(data[CONF_HOST])
+    connected = await spa.connect()
+    _LOGGER.debug("Got connected = %d", connected)
+    if not connected:
+        raise CannotConnect
+
+    # send config requests, and then listen until we are configured.
+    await spa.send_mod_ident_req()
+    await spa.send_panel_req(0, 1)
+
+    hass.loop.create_task(spa.listen())
+
+    await spa.spa_configured()
+
+    macaddr = format_mac(spa.get_macaddr())
+    model = spa.get_model_name()
+    await spa.disconnect()
+
+    return {"title": model, "formatted_mac": macaddr}
+
+
+class BalboaSpaClientFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Balboa Spa Client config flow."""
+
+    VERSION = 1
+
+    @staticmethod
+    def construct_unique_id(model_name: str, mac_addr: str) -> str:
+        """Construct the unique id from the discovery or user_step."""
+        return f"{model_name}-{mac_addr}"
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return BalboaSpaClientOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except AlreadyConfigured:
+                return self.async_abort(reason="already_configured")
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                unique_id = self.construct_unique_id(
+                    info["title"], info["formatted_mac"]
+                )
```

Comment:
The mac address is unique enough, we shouldn't add the title

https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#example-acceptable-sources-for-a-unique-id